modules/core/src/utils/flatten.js

```diff
@@ -142,3 +142,85 @@ export function flattenTypedVertices(nestedArray, {
   return start;
 }
 */
+
```

Comment:
Nit: Maybe create a new `array-utils.js`, flatten is already pretty big...

Could move fill and copy there.